### DISCUSSION TITLE
It would be useful to support the concept of a "second page" (after title) for book projects.

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#4338] [Q&A]
For the editing standards of my book project, a second page with ISBN, etc. (which seems pretty standard) is required. For the time being, I can't find a way to do this. The `index.qmd` is after the Tables of contents, etc. in a PDF file, so I can't put it there.

My solution was to create the second page in LaTeX and HTML, and use `include-before-body:` for the PDF and HTML format of the book. I used pandoc to help me create these files from a common markdown, but it seems like I'm doing what Quarto could be doing.

If you need more details, I can provide a MWE.

**Comment by fuhrmanator** [comment]
> My solution was to create the second page in LaTeX and HTML, and use `include-before-body:` for the PDF and HTML format of the book. I used pandoc to help me create these files from a common markdown, but it seems like I'm doing what Quarto could be doing.

Unfortunately, this doesn't work for HTML because this "second page" include appears on ALL pages of the HTML (it's a web site). 

It looks like I'm forced to put it in the `index.qmd` file, which has the unfortunate consequence that the (chapter) heading has to appear at the start of the "second" page.

If anyone has any ideas, I'd love to hear them.

**Reply by fuhrmanator** [reply]
I've also tried to use partials, redefining for example `title-block.html`. It's the same effect as `include-before-body:` since the title block gets loaded at the top of *every* HTML page, not just the first one.

**Reply by mcanouil** [reply]
If you have a kind of working repository for this, I can take a look and see if I can come up with an idea.

**Reply by mcanouil** [reply]
Did you try to use the template instead of partials which are only part of the template partial, _i.e._ `template.tex`? See <https://quarto.org/docs/journals/templates.html#latex-partials>.
I believe you could define your own template based on Pandoc/Quarto template to make your second title page work as expected, at least for PDF.

**Reply by fuhrmanator** [reply]
Yes, I was able to get something working with PDF (partials or includes) as I stated above. The problem was with the HTML - the template approach won't work since every chapter is essentially a web "page" that will include the second page. I realized that if I want it to work, it has to be in the index.html, at least with the current Quarto design. 

My compromise so far is to remove the the tables of contents/figures/tables from the PDF (because they appear between the cover page and the content on index.qmd) and hack the index.qmd (edit). You can see the solution at https://github.com/fuhrmanator/log210-ndc-quarto/blob/b38188f8fda0bcb44e26b6a820d1eeaf950af41a/index.qmd#L1-L7 which includes https://github.com/fuhrmanator/log210-ndc-quarto/blob/main/_deuxieme_page.qmd (the second page is included at the top of index.qmd).

It's not perfect, but it's acceptable. First, LaTeX will always place a `#` or `##` heading at the top of the page, regardless of the order of index.qmd. For example, if I try to just define the second page with no `#` at the top, the `# Preface` definition that comes after will appear at the top. I suppose this is just because the `scrbook` template always needs a header at the top. So, my hack was to used a blank `#`. To keep it from taking up 1/5 of the vertical space (the second page is supposed to fit on one printed page), I use the `\vspace` hack to move up into the blank header. This looks great in PDF, but has the downside of confusing the HTML version, making an `index.html` heading in the navigation of the HTML page:

![image](https://user-images.githubusercontent.com/7606540/219371873-05ddc4f0-d038-47e3-bd0d-78152109bdd5.png)

I even had to play around with using extra spaces after the `#` so that I didn't get `{.unnumbered .unlisted}` as a navigation entry on the left in HTML! 

For now, I can live with this. 

Interestingly, this solution makes a different result in the PDF for the index:

![image](https://user-images.githubusercontent.com/7606540/219372596-1475351a-4731-4e6d-89bb-36831945d92e.png)

It's not totally correct either, since technically there's a blank `#` with several `##` headings under it (Peface, etc.). But again, I can live with it. The constraint for the second page was not something I considered early enough. 

LaTeX has a `\frontmatter` and `\mainmatter` (which I've never used) and maybe Quarto could exploit that for the PDF solution. It would be nice if the index.qmd had a mechanism to specify the second page (akin to `\frontmatter`), and Quarto made it work for the PDF/EPUB/HTML/etc. output.

**Reply by fuhrmanator** [reply]
I also should point out that my "hack" breaks the right-side navigation of the HTML's index.html page (it doesn't appear).

**Reply by fuhrmanator** [reply]
I found that `scrbook` has some hooks for the second page (called titleback*). Maybe there are some design ideas for a Quarto solution. 

An example is at https://tex.stackexchange.com/a/100528/17868 - this solution requires that `\maketitle` be delayed after defining the second page with scrbook's titleback hooks in the `\frontmatter`

`\frontmatter` and `\maketitle` are embedded in the partials/templates of Quarto. It doesn't look feasible/easy to pull off the scrbook (Koma script) features with the current design, at least not from the index.qmd (so that the HTML also gets the second page correctly).
